### PR TITLE
Helpers 2.1

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

File: scripts/_common.sh [hunk 1]
```diff
@@ -81,24 +81,18 @@ EOT
...
             # There is a problem with version 7.3.0 and the database migration.
             # More information here: https://forum.mattermost.com/t/migrating-from-mariadb-to-postgresql-db/14194/6
-            ynh_psql_execute_as_root --sql="DELETE FROM db_migrations WHERE version=92;" --database=$db_name
+            ynh_psql_db_shell <<< "DELETE FROM db_migrations WHERE version=92;"
+
         fi
 
         # Remove the MariaDB database
-        ynh_mysql_remove_db --db_user=$mysql_db_user --db_name=$db_name
+        # FIXME ynh_mysql_drop_db && ynh_mysql_drop_user --db_user=$mysql_db_user --db_name=$db_name
```
Comment (kemenaran) [inline scripts/_common.sh:97]: 👀

File: scripts/backup [hunk 2]
```diff
@@ -6,42 +6,39 @@
...
 
-ynh_backup --src_path="$install_dir"
+ynh_backup "$install_dir"
 
 #=================================================
 # BACKUP THE APP DATA
 #=================================================
 
-ynh_backup --src_path="$data_dir" --is_big
+ynh_backup "$data_dir"
```
Comment (kemenaran) [inline scripts/backup:21]: Does that mean the data will still be excluded from light backups?

File: scripts/remove [hunk 3]
```diff
@@ -1,52 +1,47 @@
...
 # If the user attempted to upgrade from an older MariaDB-based version, but the upgrade failed,
 # the 'remove' action will automatically remove the PostgreSQL database – but doesn't know
 # that a MariaDB database is still lying around.
 # In that case, remove the MariaDB database manually.
 if mysqlshow | grep -q "^| $db_name "
 then
-	ynh_script_progression --message="Removing MariaDB database..." --weight=1
-  ynh_mysql_remove_db --db_user=$db_user --db_name=$db_name
+	ynh_script_progression "Removing MariaDB database..."
+  # FIXME ynh_mysql_drop_db && ynh_mysql_drop_user --db_user=$db_user --db_name=$db_name
```
Comment (kemenaran) [inline scripts/remove:40]: 👀

File: scripts/restore [hunk 4]
```diff
@@ -1,29 +1,23 @@
...
 chown -R $app:www-data "$install_dir"
 
 #=================================================
 # RESTORE THE DATA DIRECTORY
 #=================================================
-ynh_script_progression --message="Restoring the data directory..." --weight=4
+ynh_script_progression "Restoring the data directory..."
 
-ynh_restore_file --origin_path="$data_dir" --not_mandatory
+ynh_restore "$data_dir"
```
Comment (kemenaran) [inline scripts/restore:20]: Same thing: will the optional restore of large data dir be handled properly?

File: scripts/restore [hunk 5]
```diff
@@ -32,37 +26,35 @@ chown -R $app:www-data "$data_dir"
...
 # RESTORE THE PHP-FPM CONFIGURATION
 #=================================================
-ynh_script_progression --message="Restoring system configurations related to $app..." --weight=1
+ynh_script_progression "Restoring system configurations related to $app..."
 
-ynh_restore_file --origin_path="/etc/nginx/conf.d/$domain.d/$app.conf"
+ynh_restore "/etc/nginx/conf.d/$domain.d/$app.conf"
 
-ynh_psql_execute_file_as_root --file="./db.sql" --database=$db_name
+ynh_psql_db_shell  < "./db.sql""
```
Comment (kemenaran) [inline scripts/restore:33]: Will this execute properly as DB root (and not as the app user) ?

File: scripts/upgrade [hunk 6]
```diff
@@ -114,7 +109,7 @@ UPDATE SidebarCategories set sorting = 'manual' where sorting='';
 UPDATE SidebarCategories set sorting = 'manual' where sorting IS NULL;
 EOM
 
-    ynh_psql_execute_as_root --sql="$fix_old_version_sql" --database=$db_name
+    ynh_psql_db_shell <<< "$fix_old_version_sql"
```
Comment (kemenaran) [inline scripts/upgrade:112]: Will this properly execute as DB root - and not as the $app user?

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: scripts/backup (hunk 2 above)
Comment (alexAubin) [reply scripts/backup:21]: yes, the "--is-big" mechanism is now handled by the core, automatically for $data_dir, maybe also the log directory etc

File: scripts/restore (hunk 4 above)
Comment (alexAubin) [reply scripts/restore:20]: yes

File: scripts/upgrade (hunk 6 above)
Comment (alexAubin) [reply scripts/upgrade:105]: yes

File: scripts/restore (hunk 5 above)
Comment (alexAubin) [reply scripts/restore]: yes

**Pull request merged**